snips_nlu_metrics/metrics.py

```diff
@@ -76,40 +84,54 @@ class must inherit from `Engine`
             AVERAGE_METRICS: None,
             CONFUSION_MATRIX: None,
             METRICS: None,
-            PARSING_ERRORS: [],
+            PARSINGS: [],
         }
 
     intent_list = sorted(list(dataset["intents"]))
     global_metrics = dict()
     global_confusion_matrix = None
-    global_errors = []
+    global_parsings = []
     total_splits = len(splits)
 
     if num_workers > 1:
         effective_num_workers = min(num_workers, len(splits))
         pool = Pool(effective_num_workers)
         runner = pool.imap_unordered
     else:
+        pool = None
         runner = map
 
-    results = runner(
-        lambda split:
-        compute_split_metrics(engine_class, split, intent_list,
-                              include_slot_metrics, slot_matching_lambda),
-        splits)
+    def compute_split_metrics_wrapper(split):
+        try:
+            return compute_split_metrics(engine_class, split, intent_list,
+                                         include_slot_metrics,
+                                         slot_matching_lambda,
+                                         persist_exact_parsings)
+        except:
+            LOGGER.error('FAILURE: %s', traceback.format_exc())
+            raise RuntimeError
```

Comment:
The exception is lost here, when there is no logging.
Here is the proper way of re-raising without losing the error and its traceback:

```python
try:
    do_something()
except:
    process_error()
    raise
```

BTW, that's the only case where using `except` alone is fine.